tests/unit/rpc/handlers/AllHandlerTests.cpp

```diff
@@ -0,0 +1,213 @@
+#include "rpc/handlers/AccountObjects.hpp"
+#include "rpc/handlers/AccountOffers.hpp"
+#include "rpc/handlers/AccountTx.hpp"
+#include "rpc/handlers/BookChanges.hpp"
+#include "rpc/handlers/BookOffers.hpp"
+#include "rpc/handlers/DepositAuthorized.hpp"
+#include "rpc/handlers/Feature.hpp"
+#include "rpc/handlers/GatewayBalances.hpp"
+#include "rpc/handlers/GetAggregatePrice.hpp"
+#include "rpc/handlers/Ledger.hpp"
+#include "rpc/handlers/LedgerData.hpp"
+#include "rpc/handlers/LedgerEntry.hpp"
+#include "rpc/handlers/LedgerIndex.hpp"
+#include "rpc/handlers/MPTHolders.hpp"
+#include "rpc/handlers/NFTBuyOffers.hpp"
+#include "rpc/handlers/NFTHistory.hpp"
+#include "rpc/handlers/NFTInfo.hpp"
+#include "rpc/handlers/NFTSellOffers.hpp"
+#include "rpc/handlers/NFTsByIssuer.hpp"
+#include "rpc/handlers/NoRippleCheck.hpp"
+#include "rpc/handlers/TransactionEntry.hpp"
+#include "util/HandlerBaseTestFixture.hpp"
+#include "util/MockAmendmentCenter.hpp"
+#include "util/TestObject.hpp"
+
+#include <boost/asio/executor_work_guard.hpp>
+#include <boost/asio/io_context.hpp>
+#include <boost/asio/spawn.hpp>
+#include <boost/json/parse.hpp>
+#include <fmt/core.h>
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+#include <xrpl/protocol/AccountID.h>
+#include <xrpl/protocol/UintTypes.h>
+
+#include <string>
+
+using ::testing::Types;
+using namespace rpc;
+
+constexpr static auto Index1 = "05FB0EB4B899F056FA095537C5817163801F544BAFCEA39C995D76DB4D16F9DD";
+constexpr static auto AmmAccount = "rLcS7XL6nxRAi7JcbJcn1Na179oF3vdfbh";
+constexpr static auto Account = "rf1BiGeXwwQoi8Z2ueFYTEXSwuJYfV2Jpn";
+constexpr static auto NftID = "00010000A7CAD27B688D14BA1A9FA5366554D6ADCF9CE0875B974D9F00000004";
+constexpr static auto Currency = "0158415500000000C1F76FF6ECB0BAC600000000";
+
+using AnyHandlerType = Types<
+    AccountChannelsHandler,
+    AccountCurrenciesHandler,
+    AccountInfoHandler,
+    AccountLinesHandler,
+    AccountNFTsHandler,
+    AccountObjectsHandler,
+    AccountOffersHandler,
+    AccountTxHandler,
+    AMMInfoHandler,
+    BookChangesHandler,
+    BookOffersHandler,
+    DepositAuthorizedHandler,
+    FeatureHandler,
+    GatewayBalancesHandler,
+    GetAggregatePriceHandler,
+    LedgerHandler,
+    LedgerDataHandler,
+    LedgerEntryHandler,
+    LedgerIndexHandler,
+    MPTHoldersHandler,
+    NFTsByIssuerHandler,
+    NFTHistoryHandler,
+    NFTBuyOffersHandler,
+    NFTInfoHandler,
+    NFTSellOffersHandler,
+    NoRippleCheckHandler,
+    TransactionEntryHandler>;
+
+template <typename HandlerType>
+struct AllHandlersDeathTest : HandlerBaseTest, testing::WithParamInterface<std::string> {
+    AllHandlersDeathTest() : handler_{initHandler()}
+    {
+    }
+
+    HandlerType handler_;
+
+private:
+    HandlerType
+    initHandler()
+    {
+        if constexpr (std::is_same_v<HandlerType, AccountInfoHandler> || std::is_same_v<HandlerType, FeatureHandler>) {
+            return HandlerType{this->backend, this->mockAmendmentCenterPtr};
```

Comment:
Don't you need to initialise `mockAmendmentCenterPtr` before providing it into the handler here? Looks like it is `nullptr` now, but maybe I'm wrong.